enhancements/automated-resource-request-scaling.md

```diff
@@ -0,0 +1,635 @@
+
+   Operators could read the setting and use it for their operands, but
+   not themselves.
+
+### Locations for API setting
+
+These alternatives could apply to any of the API forms discussed in
+the previous section.
+
+1. Use a cluster-wide API
+
+   We could add a field to the Infrastructure configuration
+   resource. Many of our operators, including OLM-based operators,
+   already consume the Infrastructure API.
+
+2. Use an operator-specific API
+
+   If we choose to implement a new operator to manage scaling for the
+   control plane operators (see below), it could define a new API in a
+   CRD it owns.
+
+   Updating existing operators to read the new field and scale their
+   operands and documenting an entirely new API might be more work
+   than using the Infrastructure API that would only be useful if the
+   operator responding to the API was an add-on.
+
+### Options for declaring the base value for the CPU limit API
+
+In order for an operator to use a cluster-wide CPU resource request
+limit API, it needs to know the number of CPUs used for the
+calculation of the default settings.
+
+1. Expose the value through the Infrastructure config API.
+
+   We could publish the value as part of the status fields of the
+   Infrastructure API. This would make it easy for all operators to
+   use it, assuming the value is correct and relevant (it may not be
+   for OLM operators).
+
+   The value would only need to be set once, in the installer, and
+   could be changed in later versions without updating the source for
+   any operators. In order to change the behavior in an upgraded
+   cluster, something would have to modify the Infrastructure config
+   resource in the existing cluster.
+
+2. Expose the value as a `const` in the library where the
+   Infrastructure config API is defined.
+
+   We could publish the value only in `go` source code, using a
+   `const` in the same module where the Infrastructure API `struct` is
+   defined. Consumers of the API that import the code would be able to
+   use the value, and receive updates by updating to a new version of
+   the library.
+
+   If the value changes between releases, upgraded operators would
+   have the new value.
+
+### Dynamic vs. Static Setting
+
+It is not likely that users would need to change the setting after
+building a cluster, so we could have the installer use a value from
+the `install-config.yaml` to set a status field in whatever CRD we
+choose for the API. If we do need to support scaling later, that work
+can be described in another enhancement.
+
+### Managing settings for operators installed by cluster-version-operator
+
+The alternatives in this subsection cover the need to change the
+settings for the static manifests currently managed by the
+cluster-version-operator. These approaches would only apply to the
+operators themselves, and not their operands.
+
+1. Use a cluster profile to change the resource requests for control
```

Comment:
Recognize the support costs but this option would also allow the customization of the installed operators to exclude any not required.

Reply:
See #504 for some history of that discussion. I'll update this section with a link to that older enhancement.